### DISCUSSION TITLE
Add a private/public mode to Postal

### DISCUSSION THREAD

**Discussion by belghitioth** [postalserver/postal#2286] [Feature suggestions]
I am trying to use Postal for one of my client to manage our business needs but we don't want to display messages information as it is sensitive data.

I would like to add a private/public mode to Postal in order to not display sensitive content from emails such as attachments, plain text, plain html.

Using one new variable in the file `config/postal.defaults.yml`

`show_message_content: false` to not display sensitive data from incoming and outgoing messages

**show_message_content: false**

<img width="1255" alt="Capture d’écran 2022-12-12 à 13 48 33" src="https://user-images.githubusercontent.com/17969052/207063661-97a0d612-3269-44ab-aec1-4f16392cbe0d.png">
<img width="1242" alt="Capture d’écran 2022-12-12 à 13 50 04" src="https://user-images.githubusercontent.com/17969052/207063665-4111f96c-4cb2-4d0d-b799-9bf24b4c4cde.png">

**show_message_content: true**

<img width="1249" alt="Capture d’écran 2022-12-12 à 15 00 37" src="https://user-images.githubusercontent.com/17969052/207064271-d3a40088-9376-4bf3-9f6a-e6fdb152f75b.png">

Here is the link of the PR: https://github.com/postalserver/postal/pull/2287

**Comment by belghitioth** [comment]
@willpower232 Hello, I saw that you have commented the PR. Thanks for your message. 
What are the rules to make a feature suggestion validated ?
Thanks for your help :)

**Reply by willpower232** [reply]
I think they're all validated but its mostly up to the maintainers to choose what they want to be merged.

I'd say that whilst this feature is very neat, if people are managing messages long term then they should be looking at whatever system is sending the messages as Postal is set up to remove old messages after a while so not the best source for this information so this feature doesn't necessarily 100% fit with the point of Postal IMHO.

**Reply by belghitioth** [reply]
Hello ! The purpose of this feature is to not display messages at all because it could be sensitive information that transits from emails. Even if old messages are removed after a while, someone with the admin access could see these messages. What do you think ?

**Reply by uhlhosting** [reply]
> Hello ! The purpose of this feature is to not display messages at all because it could be sensitive information that transits from emails. Even if old messages are removed after a while, someone with the admin access could see these messages. What do you think ?

Why would you give admin to someone you do not trust...

**Reply by belghitioth** [reply]
In an business use case in a sensitive sector such as in finance or security market, we could have a customer success admin user that needs to see if all emails have been delivered and have some statistics about the delivery, such as open, bounce rate but with no rights to see the content of emails :) 
Is the use case clearer @uhlhosting  ?

**Comment by med8bra** [comment]
Nice feature @belghitioth, I find it very useful to limit  access to email content especially if the data is sensitive.

@willpower232 any hope we can have this feature soon on Postal?

**Reply by willpower232** [reply]
unfortunately it is not my say so

**Reply by belghitioth** [reply]
Hey @catphish I saw you have merged the latest PR in the repository. You think we could add this new feature on Postal ? 
It will help us a lot and help many Postal consumers to adopt the library :) 
Thanks for your help

**Reply by belghitioth** [reply]
Up, you know who could I contact ? @willpower232

**Reply by belghitioth** [reply]
Up, Hey, is there any blocking point, which maintainer could approve this ? @catphish @adamcooke

**Comment by catphish** [comment]
Shouldn't `public_mode: true` represent the new mode that hides the content?

**Reply by belghitioth** [reply]
By default, public_mode is set to true. So it keeps the same current behaviour and don't hide content.
To hide content, you have to set `public_mode: false` to not display sensitive data. https://github.com/postalserver/postal/pull/2287/files

**Reply by belghitioth** [reply]
You mean why not have a variable `private_mode: false` by default ?

**Reply by catphish** [reply]
Actually I think the problem is just that the name "public_mode" is not very clear. Can we name it something more technically descriptive, like "show_message_content"? I think it's fine for it to be true by default.

**Reply by catphish** [reply]
I was confused, because to me, "public mode" implies a mode that is suitable for the public to see (ie sensitive data is hidden). "Private mode" is equally ambiguous. Is it private because it shows everything, or because it hides things? Better to be more technical about it.

**Reply by belghitioth** [reply]
Thanks for the feedback :) 
I see what you mean. I agree "show_message_content" is more explicit, I have made the changes in the description and in the PR right [here](https://github.com/postalserver/postal/pull/2287/files)
Ok for you ?

**Comment by belghitioth** [comment]
Hello, is there any blocking point, which maintainer could approve this ? @catphish @adamcooke

**Reply by catphish** [reply]
Apologies for the delay. We have not had a lot of time to work on this project recently, but will be spending some time on these pull requests very soon.

**Reply by belghitioth** [reply]
Cool ! Happy to hear that and to see my contributions available soon :)
